ERCS/erc-7689.md

```diff
@@ -0,0 +1,256 @@
+
+## Motivation
+Executing complex data computations directly on the EVM execution layer is costly and often not economically feasible. In 2020, SmartWeave introduced an Arweave-based protocol that operates as a general lazy evaluator for data computation. By implementing smart blobs, this proposal aims to decouple state transitions (which occur on-chain, using blobs) from execution processes (which occur off-chain, using a SmartWeave instance), enhancing compute efficiency and circumventing  EVM gas costs.
+
+* Standardizing the usage of [EIP-4844](./eip-4844.md) blobs within the scope of creating execution machines on top of the blobspace
+* Utilize SmartWeave's lazy evaluation paradigm within the EVM space.
+* Extend the capabilities of lazy evaluation by using window-timegated DA
+* Utilize blobspace instead of calldata to leverage off-chain execution more cheaply
+* Write Smart Blobs in any interpreted language, or in a compiled language provided it uses WASM (JavaScript, Python, Rust, Lua, etc.)
+
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Protocol design
+The design of the proposal’s protocol is both a simple and straightforward ***computation protocol***. State transitions (transactions) are posted as [EIP-4844](./eip-4844.md) transactions by the user on the EVM network and then submitted by the user or dApp to the sequencer. The sequencer captures the EVM on-chain transaction, decodes the transaction data, performs the corresponding off-chain execution, and then indexes the state changes in the cloud (cache) after transmitting the blob data to Arweave.
+
+To simplify the expansion of the protocol design, "blobvm" refers to an execution machine deployed according to the standards of this EIP. The terms "smart blob" and "blobvm smart contract" are used interchangeably.
+
+### blobVM transactions
+On the protocol level, blobVM distinguishes between 2 types of transactions:
+
+- *type 1* : contract deployments
+- *type 2* : contract calls
+
+Transactions data structure MUST be as follows:
+
+#### Contract deployment
+```json
+{
+  "type": 1,
+  "sc": [],
+  "state":[]
+}
+```
+#### Contract call
+```json
+{
+  "type": 2,
+  "inputs": [],
+}
+```
```

Comment:
Not required for merging, but personally I'd like to see [JSON Schema](https://json-schema.org/specification) whenever an EIP describes a JSON blob.